templates/config/configmap.yaml

```
@@ -207,88 +207,113 @@ data:
         org.springframework.cloud.config.server.environment.NativeEnvironmentRepository: WARN
 
   gateway-service.yml: |
+    # Defines the root base path at which the gateway exposes all geoserver request paths
+    # We're not setting the spring.webflux.base-path because spring-cloud Gateway doesn't
+    # work properly with it. See https://github.com/spring-cloud/spring-cloud-gateway/issues/1759
+    # For example, if geoserver.base-path=/geoserver-cloud, then geoserver requests will be served
+    # under that path prefix (e.g. http://localhost:9090/geoserver-cloud/web, http://localhost:9090/geoserver-cloud/wfs, etc)
+    # Examples:
+    #geoserver.base-path:
+    #geoserver.base-path: /geoserver-cloud
+    #geoserver.base-path: /geoserver/cloud
+    # Leave it empty for no additional path, setting it to / won't work currently
+    #
+    # Please note any other prefix than the empty string will require a trailing slash in the root URL
+    # for the redirection to the web-ui to work. This is something we need to fix. For example:
+    # http://localhost:9090/geoserver-cloud will return 404, but http://localhost:9090/geoserver-cloud/ will work.
+    geoserver.base-path: {{ if not (hasSuffix "/" .Values.geoserver.ingress.baseUrl) }} {{ .Values.geoserver.ingress.baseUrl }} {{else}} {{ trimSuffix "/" .Values.geoserver.ingress.baseUrl }} {{ end }}
+
+    # Configure routes to services. See https://cloud.spring.io/spring-cloud-gateway/single/spring-cloud-gateway.html
     server:
       compression:
         enabled: true
-    # Configure routes to services. See https://cloud.spring.io/spring-cloud-gateway/single/spring-cloud-gateway.html
 
     management:
       endpoint.gateway.enabled: true # default value
       endpoints:
         enabled-by-default: true
         web.exposure.include: "*"
-        #web.exposure.include: gateway, info, health
-        
+
     spring:
       cloud:
         gateway:
+          default-filters:
+          - StripBasePath=${geoserver.base-path} #remove the base path on downstream requests
           actuator:
             verbose:
               enabled: true
           routes:
           - id: catalog
             uri: lb://catalog-service
             predicates:
-            - Path=/api/v1/**
+            - Path=${geoserver.base-path}/api/v1/**
```

Should be to be interpreted by Helm, then I suppose that's interpreted by Geoserver :-)
```suggestion
            - Path={{geoserver.base-path}}/api/v1/**
```

---

yes, it's actually used by spring boot